gammapy/astro/darkmatter/utils.py

```diff
@@ -47,17 +55,20 @@ def compute_differential_jfactor(self, ndecade=1e4):
                     rmax,
                     np.arctan(_.value / self.distance.value),
                     ndecade,
+                    self.annihilation,
                 )
                 + self.profile.integral(
                     self.distance,
                     4 * rmax,
                     np.arctan(_.value / self.distance.value),
                     ndecade,
+                    self.annihilation,
                 )
             )
             for _ in rmin.ravel()
         ]
-        jfact = u.Quantity(val).to("GeV2 cm-5").reshape(rmin.shape)
+        inegral_unit = u.Unit("GeV2 cm-5") if self.annihilation else u.Unit("GeV cm-2")
+        jfact = u.Quantity(val).to(inegral_unit).reshape(rmin.shape)
```

Comment:
```suggestion
        integral_unit = u.Unit("GeV2 cm-5") if self.annihilation else u.Unit("GeV cm-2")
        jfact = u.Quantity(val).to(integral_unit).reshape(rmin.shape)
```

Reply:
A naive question.... why in GeV?

Reply:
`GeV2 cm-5` is the standard unit for this. There's a package called clumpy (https://clumpy.gitlab.io/CLUMPY/v3.1.1/gallery.html) where you can see that too. 

Clumpy will get python binding soon. Maybe it would be of interest to also takes this into account when refactoring the dark matter stuff